exonum/src/helpers/fabric/key.rs

```diff
@@ -0,0 +1,56 @@
+use std::marker::PhantomData;
+use std::fmt;
+
+/// `Key` provides strongly typed access to data
+/// inside `Context`.
+pub struct Key<T> {
+    // These fields are public so that `config_key`
+    // macro works outside of this crate. It should be
+    // replaced with `const fn`, once it is stable.
+    #[doc(hidden)]
+    pub __name: &'static str,
+    #[doc(hidden)]
+    pub __phantom: PhantomData<T>,
+}
+
+// We need explicit `impl Copy` because derive won't work if `T: !Copy`.
+impl<T> Copy for Key<T> {}
+impl<T> Clone for Key<T> {
+    fn clone(&self) -> Self {
+        Key {
+            __name: self.__name,
+            __phantom: self.__phantom,
+        }
+    }
+}
+
+impl<T> fmt::Debug for Key<T> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "Key({:?})", self.__name)
+    }
+}
+
+impl<T> Key<T> {
+    /// Name of this key.
+    pub fn name(&self) -> &str {
+        self.__name
+    }
+}
+
+#[macro_export]
```

Comment:
Add documentation before merging, please. For example, the following doctest could reveal that the current macro implementation is not precisely correct:

```rust
#[macro_use] extern crate exonum;
use exonum::helpers::fabric::Key as ConfigKey;

fn main() {
    const KEY: ConfigKey<String> = config_key!("foo");
    // Should probably use the key somehow
}
```

Reply:
Yep, to bad `no_missing_docs` does not work for macros 🙃 

I am also not sure about naming here: the ideal names would be `Key` and `key!`, but macros are not namespaced yet, so `key!` potentially can collide with something else, and that's why I choose to name it `config_key` instead.

Perhaps this fear is unfounded, and `key!` would be a fine name? What do you all think?